server/graphql/mutations.js

```diff
@@ -39,31 +38,59 @@ const mutations = {
               slug: response.group.slug
             }
           }]
-        }]
-      })
-      .then(t => {
-        if (!t) {
-          throw new Error(`No tier found with tierId:${response.tier.id} for eventSlug:${response.event.slug} in collectiveSlug:${response.group.slug}`);
-        }
-        tier = t;
-      })
+        })
+        .then(ev => {
+          if (!ev) {
+            throw new Error(`No event found with slug: ${response.event.slug} in collective: ${response.group.slug}`)
+          }
+          event = ev;
+        })
```

Comment:
this is hard to follow. I would split the case for "interested" in a different function. It would be easier to read; if status is interested, go this way and ignore everything that follows (finding a tier, check availability, pay, etc.)